doc/rm/otbn_coding_style.md

```diff
@@ -0,0 +1,232 @@
+
+Generally, OTBN routines should be written so that they finish in `ret`.
+Use of `ecall` should be restricted to thin wrapper files which serve as an interface for the more substantial assembly files, usually just reading one or two arguments and then calling subroutines from the other files.
+
+### Comments
+
+Because of toolchain limitations, the `//` syntax is not available for OTBN.
+Use the `/* */` syntax instead.
+
+### Labels
+
+It is not required to use an `L_` prefix for internal labels.
+Labels should not be indented.
+
+### Operand Alignment
+
+***Operands should be aligned within blocks of OTBN code.***
+
+The exact spacing betweeen mnemonic and operand is not important, as long as it is consistent within the block.
+
+Example:
+```S
+  /* Correct */
+  bn.add  w4, w7, w10
+  bn.addc w5, w8, w11
+  bn.sel  w2, w2, w3, C
+
+  /* Correct */
+  bn.add        w4, w7, w10
+  bn.addc       w5, w8, w11
+  bn.sel        w2, w2, w3, C
+
+  /* Wrong */
+  bn.add w4, w7, w10
+  bn.addc w5, w8, w11
+  bn.sel w2, w2, w3, C
+```
+
+### Register and flag group clobbering
+
+***Always document which registers and flag groups are clobbered by an OTBN "function", and which flags have meaning.***
+
+OTBN subroutines should always document the registers and flag groups whose values they overwrite, including those used for output.
+In addition to Doxygen-style `@param[in]` and `@param[out]` notations, OTBN subroutines should also state whether the flags have meaning at the end of the subroutine.
+If a subroutine jumps to another subroutine that clobbers additional registers or flag groups, these additional names should be added to the caller's list.
+
+Example:
+```S
+/**
+ * Compute the sum of two 2048-bit numbers.
+ *
+ *   Returns C = A + B.
+ *
+ * Flags: When leaving this subroutine, the C flag of FG0 depends on the
+ *        addition.
+ *
+ * @param[in]  [w10:w3]: A, first 2048-bit operand
+ * @param[in]  [w18:w11]: B, second 2048-bit operand
+ * @param[in]  w31: all-zero
+ * @param[out] [w26:w19]: C, result
+ *
+ * clobbered registers: x9 to x13, w19 to w28
+ * clobbered flag groups: FG0
+ */
+wide_sum:
+  /* Prepare temporary registers. */
+  li       x9, 3
+  li       x10, 11
+  li       x11, 19
+  li       x12, 27
+  li       x13, 28
+
+  /* Clear flags. */
+  bn.add   w31, w31, 0
+
+  /* Addition loop. */
+  loopi    8, 4
+    /* w27 <= A[i] */
+    bn.movr  x12, x9++
+    /* w28 <= B[i] */
+    bn.movr  x13, x10++
+    /* w27 <= w27 + w28 */
+    bn.addc  w27, w27, w28
+    /* C[i] <= w27 */
+    bn.movr  x11++, x12
+
+  ret
+```
+
+## Secure Coding for Cryptography
+
+The following guidelines address cryptography-specific concerns for OTBN assembly.
+
+### Copying register values
+
+Prefer `bn.mov <wrd>, <wrs>` to `bn.addi <wrd>, <wrs>, 0` when copying data between registers.
+Because `bn.addi` passes data through the ALU, it will strip off integrity protection, while `bn.mov` will copy the integrity protection bits.
```

Comment:
As a side-note: this might be another simple thing for your proposed linter to check for.